### DISCUSSION TITLE
Tablet not working in Android

### DISCUSSION THREAD

**Discussion by Neon095** [ppy/osu#16494] [Ideas] [answered]
I tried to connect my tablet (Xp-pen star g-430s) to my android device and it doesn't work in lazer. I have seen that there are some draw applications that let you actually use the tablet, and i was thinking if that feature could be implemented to lazer too.

**Comment by smoogipoo** [comment, marked as answer]
https://github.com/ppy/osu/discussions/15188

**Reply by Neon095** [reply]
Log of what? Because is not a problem is an unimplemented feature

**Reply by peppy** [reply]
And you know that for a fact? We ask for logs for a reason. If you aren't going to provide them then please don't post in the first place.

**Reply by Neon095** [reply]
Ok sry. With the tablet connected?

**Reply by peppy** [reply]
Yes please.

**Reply by Neon095** [reply]
[performance.log](https://github.com/ppy/osu/files/7892599/performance.log)

**Reply by Neon095** [reply]
Sorry for the delay, here are the logs:
[network.log](https://github.com/ppy/osu/files/7892450/network.log)
[runtime.log](https://github.com/ppy/osu/files/7892454/runtime.log)

**Reply by bdach** [reply]
OTD will not work without .NET 5, which is a no-go for mobile (currently stuck on netstandard), but I'm not sure why the tablet wouldn't work if the Android OS is generally handling it fine. May need https://github.com/ppy/osu-framework/pull/4989 to go in, but I'm not really sure why (I'd expect the tablet to be handled as `InputSourceType.Touchpad` already).

I don't really have good means to test this myself, as I'm not sure how people are even connecting other USB devices to their phones...?

**Reply by Neon095** [reply]
No i was wrong, actually i have seen that the tablet does not work in any app, the cursor get stock in the center of the screen

**Reply by Neon095** [reply]
Also, we connect the usb with an adapter

**Reply by bdach** [reply]
If the tablet doesn't work with any app at all then safe to say that the android OS just doesn't support it, at which point there's not much we can do. Driver development is not in scope of this repository.
